src/Workspaces/Core/Portable/ReassignedVariable/AbstractReassignedVariableService.cs

```diff
@@ -0,0 +1,326 @@
+            }
+
+            void ProcessParameter(TParameterSyntax parameterSyntax)
+            {
+                var parameter = semanticModel.GetDeclaredSymbol(parameterSyntax, cancellationToken) as IParameterSymbol;
+                if (IsSymbolReassigned(parameter))
+                    result.Add(syntaxFacts.GetIdentifierOfParameter(parameterSyntax).Span);
+            }
+
+            void ProcessVariable(TVariableSyntax variable)
+            {
+                var local = semanticModel.GetDeclaredSymbol(variable, cancellationToken) as ILocalSymbol;
+                if (IsSymbolReassigned(local))
+                    result.Add(GetIdentifierOfVariable(variable).Span);
+            }
+
+            void ProcessSingleVariableDesignation(TSingleVariableDesignationSyntax designation)
+            {
+                var local = semanticModel.GetDeclaredSymbol(designation, cancellationToken) as ILocalSymbol;
+                if (IsSymbolReassigned(local))
+                    result.Add(GetIdentifierOfSingleVariableDesignation(designation).Span);
+            }
+
+            bool IsSymbolReassigned([NotNullWhen(true)] ISymbol? symbol)
+            {
+                // Note: we don't need to test range variables, as they are never reassignable.
+                if (symbol is not IParameterSymbol and not ILocalSymbol)
+                    return false;
+
+                if (!symbolToIsReassigned.TryGetValue(symbol, out var reassignedResult))
+                {
+                    reassignedResult = symbol is IParameterSymbol parameter
+                        ? ComputeParameterIsAssigned(parameter)
+                        : ComputeLocalIsAssigned((ILocalSymbol)symbol);
+                    symbolToIsReassigned[symbol] = reassignedResult;
+                }
+
+                return reassignedResult;
+            }
+
+            bool ComputeParameterIsAssigned(IParameterSymbol parameter)
+            {
+                if (!TryGetParameterLocation(parameter, out var parameterLocation))
+                    return false;
+
+                var methodOrProperty = parameter.ContainingSymbol;
+
+                // If we're on an accessor parameter. Map up to the matching parameter for the property/indexer.
+                if (methodOrProperty is IMethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet } method)
+                    methodOrProperty = method.AssociatedSymbol as IPropertySymbol;
+
+                if (methodOrProperty is not IMethodSymbol and not IPropertySymbol)
+                    return false;
+
+                if (methodOrProperty.DeclaringSyntaxReferences.Length == 0)
+                    return false;
+
+                // Potentially a reference to a parameter in another file.
+                var methodOrPropertyDeclaration = methodOrProperty.DeclaringSyntaxReferences.First().GetSyntax(cancellationToken);
+                if (methodOrPropertyDeclaration.SyntaxTree != semanticModel.SyntaxTree)
```

Comment:
also might be good to have a test with this case, I didn't see any multi file tests

Reply:
so i'll comment this, but this is a general paranoia check normally for error sitautions.  Specifically, when you have duplicate symbols (for example duplicate methods), in general you can't be certain what the compiler will do.  It sometimes happens that they just merge the duplicates into a single symbol with multiple locations.  So, when checking locations like this, we just try to be resilient just in case soemthing like that happens.  

i'll add a test for this sort of thing.